cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -418,6 +418,15 @@ private void handleProgramEvent(ProgramRunId programRunId, ProgramRunStatus prog
         recordedRunRecord = appMetadataStore.recordProgramStopping(programRunId, messageIdBytes, stoppingTsSecs,
                                                                    terminateTsSecs);
         writeToHeartBeatTable(recordedRunRecord, stoppingTsSecs, programHeartbeatTable);
+        // if the stop requested was graceful, emit the total number of graceful stops metric
+        long gracefulShutdownSecs = terminateTsSecs - stoppingTsSecs;
+        if (gracefulShutdownSecs > 0) {
```

Comment:
Why need this check?

Reply:
Yeah, this condition shouldn't occur

Reply:
Actually we need this condition. If graceful shutdown period is 0 it means it was an immediate kill and not partially graceful.

Reply:
Ah okay, I was classifying it into 3 groups - fully graceful which means we will wait for the program to finish on it's own, partially meaning we will wait for sometime and immediate kill when we don't want to wait at all.
It does make sense to just say it was graceful or not graceful combining the last two groups. Made relevant changes, thanks